cli/src/stake.rs

```diff
@@ -1892,14 +1921,26 @@ pub fn process_split_stake(
             return Err(CliError::BadParameter(err_msg).into());
         }
 
-        let minimum_balance =
-            rpc_client.get_minimum_balance_for_rent_exemption(StakeState::size_of())?;
+        if !force {
+            let minimum_balance = rpc_client
+                .get_minimum_balance_for_rent_exemption(StakeState::size_of())?
+                + rpc_client.get_stake_minimum_delegation()?;
+            if lamports < minimum_balance {
```

Comment:
Really sorry, I'm just noticing this now... this check isn't correct in all cases, since you could have a destination account with the rent-exempt minimum, and you split just the minimum delegation into that account.

Since you're already fetching the account at `split_stake_account_address`, you could get its current lamports, and then do this check instead:

```
let required_rent_lamports = rpc_client.get_minimum_balance_for_rent_exemption(...)?.saturating_sub(current_lamports);
let minimum_delegation = rpc_client.get_stake_minimum_delegation()?;
if lamports < required_rent_lamports + minimum_delegation {
...
}
```

This would be a nice check to update since it's pretty common to seed the destination account with lamports pre-split

Reply:
Is the destination account not always a newly created account as part of the split-stake? Can it be an existing sytem account?

Reply:
You can use any account that has 200 empty bytes allocated and assigned to the stake program

Reply:
> The minimum split amount will be based on the source's state. If the source is staked, the minimum split amount is the minimum delegation. If the source is not staked, then there is no minimum (or maybe 1 lamport until some feature gates are activated).

Correct, I didn't want to bring this up for fear of adding extra confusion, but yeah you can split pretty much any amount from an unstaked account, but it's also possible to withdraw + create new account.

> This split amount effectively satisfies the destinations minimum balance check, assuming it also previously satisfies the rent exempt minimum check.

Sorry, I'm not sure I understand what you're asking. In our case, we're assuming always the the source is staked, so `lamports` must be at least `minimum_delegation` + `additional_lamports_required_for_rent_exemption`. If the destination has 0 lamports, then `additional_lamports_required_for_rent_exemption` will be the full rent-exemption, otherwise it can go down to `0` if it already has that amount in it. With the code as it is, `lamports` must always be `minimum_delegation` + `full_rent_exemption`, and will improperly fail if the destination already has the rent-exemption lamports in it and you try to split `minimum_delegation`.

Reply:
Ok cool! Glad that we're on the same page :smile:

Reply:
Hey Guys sorry this slipped under the radar for a bit. Re the using an existing account, wouldn't this fail on the check on line 1912:
```let err_msg = if stake_account.owner == stake::program::id() {
                format!("Stake account {split_stake_account_address} already exists")
} else {
                format!(
                    "Account {split_stake_account_address} already exists and is not a stake account"
                )
            };```

implying you can't use an existing account?

Reply:
Oh yeah, that's a good point! I put in https://github.com/solana-labs/solana/issues/32735 to address that, so we can do it in a separate PR. Either way, this should still do the right thing with the balance calculations for when that's fixed. Does that work?